ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DownloadHeadersStep.java

```diff
@@ -99,7 +99,6 @@ private CompletableFuture<List<BlockHeader>> downloadHeaders(final SyncTargetRan
               range.getStart().getNumber(),
               headerRequestSize,
               metricsSystem)
-          .assignPeer(range.getSyncTarget())
```

Comment:
For POS this is safe as the range start block number is used as the minimum required block. For non-pos chains not sure whether this is safe think might be an issue.

If you only skipped this when it's POS then I think it would be better.

Reply:
Thinking through this more, it's already possible that other nodes might not have this header and we will fail to download the block. So I think this change is fine.